openmdao/core/problem.py

```diff
@@ -1470,6 +1471,21 @@ def check_totals(self, of=None, wrt=None, out_stream=_DEFAULT_OUT_STREAM, compac
 
         # TODO: Once we're tracking iteration counts, run the model if it has not been run before.
 
+        if wrt is None:
+            wrt = list(self.driver._designvars)
+            if wrt is None:
+                raise RuntimeError("Driver is not providing any design variables "
+                                   "for compute_totals.")
+
+        if of is None:
+            of = list(self.driver._objs)
+            of.extend(self.driver._cons)
+            lcons = [n for n, meta in self.driver._cons.items()
+                     if ('linear' in meta and meta['linear'])]
+            if of is None:
```

Comment:
Very minor, but I would put the if check before the `lcons = ...` line above.

Reply:
Fixed